packages/@groupby/elements-gb-tracker-plugin/src/gb-tracker-plugin.ts

```diff
@@ -0,0 +1,98 @@
+import { Plugin, PluginMetadata, PluginRegistry } from '@groupby/elements-core';
+import { BEACON_SEARCH, BeaconSearchPayload } from '@groupby/elements-events';
+// eslint-disable-next-line import/no-unresolved
+import { AutoSearchEvent } from 'gb-tracker-client/models';
+// eslint-disable-next-line import/no-unresolved
+import { GbTracker } from 'gb-tracker-client/slim-es';
+
+/**
+ * This plugin is responsible for exposing an [[GbTracker]] instance to Core.
+ */
+export default class GbTrackerPlugin implements Plugin {
+  get metadata(): PluginMetadata {
+    return {
+      name: 'gb-tracker',
+      depends: ['dom_events'],
+    };
+  }
+
+  /**
+   * The value that the [[GbTrackerPlugin]] exposes to the Core entity.
+   */
+  gbTracker: ReturnType<typeof GbTracker>;
+
+  /**
+   * A reference to the registry of plugins for later use.
+   */
+  core: PluginRegistry;
+
+  /**
+   * Name of the events plugin.
+   */
+  eventsPluginName = 'dom_events';
+
+  /**
+   * The [[GbTrackerPlugin]] constructor instantiates an instance of the
+   * [[GbTracker]] client and attaches it to this plugin's `gbTracker`
+   * property.
+   *
+   * @param options The options for instantiating the GbTracker client.
+   */
+  constructor(options: TrackerPluginOptions) {
+    this.triggerSearchBeacon = this.triggerSearchBeacon.bind(this);
+
+    this.gbTracker = new GbTracker(options.customerId, options.area, options.overridePixelUrl);
+  }
+
+  /**
+   * Returns this plugin's instance of the [[GbTracker]] client.
+   */
+  register(plugins: PluginRegistry): ReturnType<typeof GbTracker> {
+    this.core = plugins;
+    return this.gbTracker;
+  }
+
+  /**
+   * Sets event listeners for beacon events. Also calls the GbTracker method
+   * `autoSetVisitor()` to allow for future beacon events to be registered.
+   */
+  ready(): void {
+    this.core[this.eventsPluginName].registerListener(BEACON_SEARCH, this.triggerSearchBeacon);
+
+    this.gbTracker.autoSetVisitor();
+  }
+
+  /**
+   * Unregister event listeners for beacon events.
+   */
+  unregister(): void {
+    this.core[this.eventsPluginName].unregisterListener(BEACON_SEARCH, this.triggerSearchBeacon);
+  }
+
+  /**
+   * Triggers a Search beacon. Intended for use as an event listener.
+   *
+   * @param event The event containing search tracking data.
+   */
+  triggerSearchBeacon(event: CustomEvent<BeaconSearchPayload>): void {
+    const payload: AutoSearchEvent = {
+      search: {
+        id: event.detail.results.id,
+        origin: event.detail.origin,
+      },
+    };
+    this.gbTracker.sendAutoSearchEvent(payload);
+  }
+}
+
+/**
+ * The options used to instantiate a [[GbTrackerPlugin]].
+ */
+export interface TrackerPluginOptions {
```

Comment:
If this fits in MVP, you should consider accepting a tracker instance. I'm sure some clients will have already instantiated their own if their site isn't completely on Elements or if they're doing anything complex.

Reply:
How will we know how to interact with a client tracker instance? I'm sure we talked about this and I don't remember the nuances. The idea with firing events was to avoid that whole issue could be sidestepped and we can allow clients to use their own trackers by simply listening to events. Or would they pass callbacks for interacting with their tracker?

Reply:
We assume that it's an external gb-tracker instance. Instead of creating our own, we allow them to pass one in. It would be a breach of contract if they pass in anything that's not a gb-tracker.

Reply:
Ah, I misunderstood. I think we did agree to do that, but I'm going to push it to a follow-up ticket.